implementing Interfaces

The docs state:
> `#[implements(ce)]` - Implements the given interface on the class. Can be used multiple times. ce must be a valid Rust expression when it is called inside the `#[php_module]` function.

How do I use this to implement an interface provided by the standard library of PHP? For example, I want to implement [`ArrayAccess`](https://www.php.net/manual/en/class.arrayaccess.php) on my own class.

My current approach is this:
```rs
#[php_class]
#[implements(ext_php_rs::zend::class::ClassEntry::try_find("ArrayAccess").unwrap())]
pub struct Test {
}
```
But the `class` type seems to be private:
```
error[E0603]: module `class` is private
   --> src/lib.rs:205:1
```

If I get this to work, how do I actually implement the methods? Does having the appropriate method names suffice (putting renaming of methods aside)?

If this has a solution, I'd love to contribute this to the docs. I really like this library!

--- a/guide/src/macros/classes.md
+++ b/guide/src/macros/classes.md
@@ -88,10 +88,11 @@ The following example implements [`ArrayAccess`](https://www.php.net/manual/en/c
 # #![cfg_attr(windows, feature(abi_vectorcall))]
 # extern crate ext_php_rs;
 use ext_php_rs::prelude::*;
-use ext_php_rs::{exception::PhpException, zend::ce, types::Zval};
+use ext_php_rs::{exception::PhpResult, types::Zval, zend::ce};
 
 #[php_class]
 #[implements(ce::arrayaccess())]
+#[derive(Default)]
 pub struct EvenNumbersArray;
 
 /// Returns `true` if the array offset is an even number.
@@ -125,4 +126,9 @@ impl EvenNumbersArray {
         Err("Setting values is not supported".into())
     }
 }
+# #[php_module]
+# pub fn get_module(module: ModuleBuilder) -> ModuleBuilder {
+#     module
+# }
+# fn main() {}
 ```

--- a/guide/src/macros/classes.md
+++ b/guide/src/macros/classes.md
@@ -85,6 +85,11 @@ pub fn throw_exception() -> PhpResult<i32> {
 To implement an interface, use `#[implements(ce)]` where `ce` is an expression returning a `ClassEntry`.
 The following example implements [`ArrayAccess`](https://www.php.net/manual/en/class.arrayaccess.php):
 ```rust,no_run
+# #![cfg_attr(windows, feature(abi_vectorcall))]
+# extern crate ext_php_rs;
+use ext_php_rs::prelude::*;
+use ext_php_rs::{exception::PhpException, zend::ce, types::Zval};
+
 #[php_class]
 #[implements(ce::arrayaccess())]
 pub struct EvenNumbersArray;

--- a/guide/src/macros/classes.md
+++ b/guide/src/macros/classes.md
@@ -79,3 +79,45 @@ pub fn throw_exception() -> PhpResult<i32> {
 # }
 # fn main() {}
 ```
+
+## Implementing an Interface
+
+To implement an interface, use `#[implements(ce)]` where `ce` is an expression returning a `ClassEntry`.
+The following example implements [`ArrayAccess`](https://www.php.net/manual/en/class.arrayaccess.php):
+```rust,no_run
+#[php_class]
+#[implements(ce::arrayaccess())]
+pub struct EvenNumbersArray;
+
+/// Returns `true` if the array offset is an even number.
+/// Usage:
+/// ```php
+/// $arr = new EvenNumbersArray();
+/// var_dump($arr[0]); // true
+/// var_dump($arr[1]); // false
+/// var_dump($arr[2]); // true
+/// var_dump($arr[3]); // false
+/// var_dump($arr[4]); // true
+/// var_dump($arr[5] = true); // Fatal error:  Uncaught Exception: Setting values is not supported
+/// ```
+#[php_impl]
+impl EvenNumbersArray {
+    pub fn __construct() -> EvenNumbersArray {
+        EvenNumbersArray {}
+    }
+    // We need to use `Zval` because ArrayAccess needs $offset to be a `mixed`
+    pub fn offset_exists(&self, offset: &'_ Zval) -> bool {
+        offset.is_long()
+    }
+    pub fn offset_get(&self, offset: &'_ Zval) -> PhpResult<bool> {
+        let integer_offset = offset.long().ok_or("Expected integer offset")?;
+        Ok(integer_offset % 2 == 0)
+    }
+    pub fn offset_set(&mut self, _offset: &'_ Zval, _value: &'_ Zval) -> PhpResult {
+        Err("Setting values is not supported".into())
+    }
+    pub fn offset_unset(&mut self, _offset: &'_ Zval) -> PhpResult {
+        Err("Setting values is not supported".into())
+    }
+}
+```

